### PR TITLE
deps: update chai-http to latest

### LINKED ISSUE #951: (deps) Update chai-http to latest
Our integration tests are stuck on `chai-http` version 1.0.0 due to breaking changes introduced in https://github.com/chaijs/chai-http/commit/3083ab17bb371f9e6ec3076577c5778e05787b5f.  Basically, the library removed the `.then()` and `.catch()` methods, preferring to use `.end(err, res)` instead.

All of our integration tests need to be updated to accommodate this change.  That will be a big undertaking, but potentially worth it to stay up to date with security fixes and performance improvements in the underlying libraries.

### PR DESCRIPTION
**Pull request by jniles**
Updates chai-http to the latest version.  This took forever because of
the number of changes to chai-http, but turned out not to be as painful
as before version 4.

Closes #951.

### CODE REVIEW

**Comment by jniles** [comment]
bors try

**Comment by jniles** [comment]
@lomamech, can I get a review?

File: test/integration/debtorGroups.js [hunk 1]
```diff
@@ -312,7 +311,7 @@ describe('(/debtor_groups) The debtor groups API', () => {
     });
 
     it('GET /debtor_groups/:uuid/invoices returns only balanced invoices for a debtor group', () => {
-      return agent.get(`/debtor_groups/${debtorGroup.uuid}/invoices?balanced=1`)
+      return agent.get(`/debtor_groups/${debtorGroup.uuid}/invoices`).query({ balanced : 0 })
```
Comment (mbayopanda) [inline test/integration/debtorGroups.js]: The test is for balanced invoices (`balanced = 1`) but you send `balanced = 0`, does it correct ?

Comment (jniles) [reply test/integration/debtorGroups.js]: Hey, you are right!  Good catch.  Let me change that real quick.

File: test/integration/fillForm.js [hunk 2]
```diff
@@ -54,18 +55,19 @@ describe('(/fill_form) Fill Form', () => {
...
+        value : 3,
+      }],
       collectorId : 3,
       searchDateFrom : {
         dateSurvey : '2018-12-31',
       },
       searchDateTo : {
         dateSurvey : '2019-12-31',
       },
+      test :  [1, 2, 3],
```
Comment (jniles) [inline test/integration/fillForm.js]: Whoops, these should also be removed.

**Comment by jniles** [comment]
@mbayopanda thanks for the review.  Changes have landed.

**Review by mbayopanda — APPROVED** [review]
LGTM 👍 

bors r+

**Pull request merged**